Illegal tech audit

bloons3 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Closes https://github.com/BeeStation/BeeStation-Hornet/issues/3755

Removes illegal technology from most items that spawn on station, are useless and require no tradeoff to destroy in research, are common, or can be crafted.

- Syndicate Throwing Weapons - Start in contraband locker, also common in maint
- Flamethrower - Can be crafted
- Ammo magazines - Trash/common, also found on ruins. No 'cost' to burn them for illegal tech
- Mosin Ammo - Can be ordered from Cargo
- Syndicate Soap - Cheap throwaway item
- Syndicate Toolbox - Found in ruins
- Syndicate Playing Cards - Common in maint
- Syndicate Briefcase - Byproduct of buying money with TC, is identical to a regular secure briefcase
- Surplus Rifle - Can be ordered from Cargo
- Fake Nuclear Authentication Disk - Allows meta of checking the disk as a non-captain, also spawns roundstart in the Chapel of a map
- Smuggler Satchel - The regular version found in the floor cannot be deconstructed for illegal tech, but the uplink item can despite only starting with a crowbar and floor tile. Allows meta-info that it was purchased from an uplink
- Grenade Launcher - Created a special subtype for Armory which prevents illegal tech
- Armory Contraband Stechkin - converted into a version that doesn't award illegal tech

- ~~Boxes and storage items themselves, since most are just normal boxes/bags~~
- This PR should be followed by another that allows the illegal tech research to look "inside" of the storage for additional items to tag as illegal technology.




## Why It's Good For The Game
This makes it harder to get illegal tech roundstart. 

For the past three rounds, I have watched the RD tide into Armory to get the grenade launcher to sacrifice for illegal technology. 

No more.

Illegal tech requires a tradeoff of using contraband seized from a traitor or destroying it, not from walking into the Armory at the start of the round.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
![image](https://user-images.githubusercontent.com/3241376/112737195-91a57100-8f26-11eb-974a-2404734a1702.png)

## Changelog
:cl:
tweak: Removed the ability to unlock illegal tech using common items, see #3988 for full list
mapping: Armory starts with official NT brand Grenade Launchers instead of stolen Syndicate ones
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

bloons3 committed:
no throwing

- spawn in armory locker

bloons3 committed:
flamethrower

- can be crafted on station

bloons3 committed:
most basic ammo magazines

- illegal tech = sacrifice something *useful*

bloons3 committed:
mosin ammo

- purchased via cargo

bloons3 committed:
soap

- it's soap

bloons3 committed:
syndicate toolbox

- relatively common on ruins

bloons3 committed:
syndi cards

- maint common

bloons3 committed:
syndi briefcase

- waste product of getting money, doesn't require a tradeoff

bloons3 committed:
fake nuclear disk

- provides a meta way to tell that it isn't the real disk
- also found in pubby chapel

bloons3 committed:
grenade launcher

- roundstart free illegal tech in the armory is bad

bloons3 committed:
tab :)

bloons3 committed:
more variety

no longer gives illegal tech, so drop a minor item in maint

bloons3 committed:
surplus rifle

- ordered via cargo

bloons3 committed:
Smuggler's Satchel

- found in floor, this item allows meta-info that it was purchased from an uplink

Minelol76 commented:
B-but

Muh energy bow for syndie cards

OceanFish1 commented:
What's wrong with the ones on ruins? Syndicate toolbox and ammo for syndicate weapons are actually not syndicate or what?

ItsTisme commented:
I wouldnt see the problen with syndie soap. Its exclusive to uplink so why not have it unlock illegal tech?

WhyIsCaeciliusTaken commented:
> I wouldnt see the problen with syndie soap. Its exclusive to uplink so why not have it unlock illegal tech?

Hmm yes the hyper advanced syndicate technology known as "painting soap red"

ItsTisme commented:
> > I wouldnt see the problen with syndie soap. Its exclusive to uplink so why not have it unlock illegal tech?
> 
> Hmm yes the hyper advanced syndicate technology known as "painting soap red"

Ah sorry, yeah soap isaht that advanced

especialy compared to the marvals of inovation like "how to fold paper", a bottle of glue, a baloon and a (ironcly) red and black painted toolbox

ivanmixo commented:
damn syndicate and their soap technology

Effemu commented:
why ruins though 😕

bloons3 commented:
> What's wrong with the ones on ruins? Syndicate toolbox and ammo for syndicate weapons are actually not syndicate or what?

Illegal tech requires a tradeoff of using contraband seized from a traitor or destroying it.

You can have an emag, or burn it for illegal technology. Burning an empty stetchkin magazine costs you nothing and gives you very valuable illegal technology.

Along the same lines, a Syndicate toolbox is exactly the same as a regular toolbox and there's no reason to value it instead of burning it for illegal technology.

utilbird reviewed (dismissed):
I'm a bit torn on items obtainable at ruins like toolboxes and ammo. They're not really worth the effort unless someone has a gimmick with illegal tech in mind, but I can still see the reasoning behind it so 🤷. Everything else is alright

bloons3 commented:
> I'm a bit torn on items obtainable at ruins like toolboxes and ammo. They're not really worth the effort unless someone has a gimmick with illegal tech in mind, but I can still see the reasoning behind it so 🤷. Everything else is alright

You can get a syndicate hardsuit out of the space ruin with a safe, and that's definitely _worth it_ for illegal tech, since on one hand the hardsuit is nice by itself and very useful but on the other hand you can burn it for very useful illegal technology.

ItsTisme commented:
Its kinda sad though if a traitor actuely wants illegal tech for his traitoring he now had to invest into more expensive items which is lil dumb imo.

jupyterkat commented:
for a station that huge and with randomized maints, at least some items of illegal tech must be there. c'mon there's fricking tableport armor in maint.

bloons3 commented:
> for a station that huge and with randomized maints, at least some items of illegal tech must be there. c'mon there's fricking tableport armor in maint.

Ah yes, this empty and crumpted bag of Syndicate Potato Chips I found in maint will give the research division incredible insights into illegal technology!

jupyterkat commented:
yes, or these nanometer thin cards that can embed into people and cut them

jupyterkat commented:
or frickin shurikens that cannot be produced anywhere but the syndicate using syndicate technology

jupyterkat commented:
Or bags that can fit between tiny crevices with items that have wayy larger volumes inside

jupyterkat commented:
People breaking into armories is purely an ic issue and it’s why wardens exist

bloons3 commented:
> Or bags that can fit between tiny crevices with items that have wayy larger volumes inside

![image](https://user-images.githubusercontent.com/3241376/112812396-b4c54300-9042-11eb-97ac-634fce75e315.png)

> People breaking into armories is purely an ic issue and it’s why wardens exist

You don't deserve illegal technology for walking into armory at the start of the round or rushing to the Bar.

jupyterkat commented:
Gives another reason for people to break into the armory

jupyterkat commented:
> > Or bags that can fit between tiny crevices with items that have wayy larger volumes inside
> 
> ![image](https://user-images.githubusercontent.com/3241376/112812396-b4c54300-9042-11eb-97ac-634fce75e315.png)
> 
> > People breaking into armories is purely an ic issue and it’s why wardens exist
> 
> You don't deserve illegal technology for walking into armory at the start of the round or rushing to the Bar.

If the spaces were that large then why can’t other backpacks or duffelbags can’t fit innit

bloons3 commented:
For reasons of game balance.

Smuggler's Satchels used to persist between rounds. Guess why that was removed.

jupyterkat commented:
they must use special technology to make them fit in then

jupyterkat commented:
I don’t agree with removing the chance of getting tator tech from all station things. At least something must be. Make it randomly spawn in maint if you hate armory break ins so much.

spockye commented:
i wouldnt say that i  have tided into the armoury. i always ask the warden, hos or captain befor taking it.

PowerfulBacon reviewed (dismissed):
Good but ammo should be illegal (Nukie / traitor throws away empty magazine, get illegal tech)

bloons3 committed:
armory roundstart stechkin

- makes non-illegal tech for roundstart stechkin

bloons3 commented:
> Good but ammo should be illegal (Nukie / traitor throws away empty magazine, get illegal tech)

There's no cost/tradeoff to burning an empty magazine for illegal tech. A gun yes, but a magazine not as much.

DatBoiTim commented:
I feel like ruins should be a pass, as it gives a "desire" for exploration. Roundstart armory/maints? Yeah not quite.

bloons3 committed:
gives kilo a grenade launcher

- kilo needs love too

bloons3 committed:
prevents bags/boxes being deconned instead of the items inside

bloons3 committed:
fix box items giving illegal tech and not their contents

bloons3 commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1139,7 +1139,8 @@
 	for(var/path in GLOB.uplink_items)
 		var/datum/uplink_item/UI = new path
-		if(!UI.item || !UI.illegal_tech || istype(UI, /obj/item/storage))
+		if(!UI.item || !UI.illegal_tech || ispath(UI.item, /obj/item/storage))
```
Old

An empty duffel bag ![image](https://user-images.githubusercontent.com/3241376/112898326-4dd97580-90a6-11eb-93d9-70409e5be87e.png)

An empty box 
![image](https://user-images.githubusercontent.com/3241376/112898415-6b0e4400-90a6-11eb-929a-beb22a7038ad.png)

---
New 

![image](https://user-images.githubusercontent.com/3241376/112900177-ba557400-90a8-11eb-9410-e42bedb2a5aa.png)

Box items (not their contents) no longer give illegal tech

![image](https://user-images.githubusercontent.com/3241376/112900326-f4bf1100-90a8-11eb-8489-1b9076877941.png)

bloons3 committed:
fix a regression with briefcase teleporter

- briefcase teleporter is a storage path object, but _should_ be deconned as illegal tech
- target boxes and bags of items instead

bloons3 committed:
missing closing parenthesis

Sinestia commented:
> 
> 
> I feel like ruins should be a pass, as it gives a "desire" for exploration. Roundstart armory/maints? Yeah not quite.

They still have a decent amount I think

just less of the guaranteed stuff since syndicate toolboxes are everywhere. I'd prefer they werent removed but... it's not that much of an issue.

utilbird commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1139,7 +1139,7 @@
 	for(var/path in GLOB.uplink_items)
 		var/datum/uplink_item/UI = new path
-		if(!UI.item || !UI.illegal_tech)
+		if(!UI.item || !UI.illegal_tech || ispath(UI.item, /obj/item/storage/box) || ispath(UI.item, /obj/item/storage/backpack))
```
It'd be better make a blacklist typecache instead.
i.e
```
var/static/list/blacklist = typecacheof(list(
	/obj/item/storage/box,
	/obj/item/storage/backpack
	))

PowerfulBacon replied:
Just disable illegal tech on the things that give boxes

bloons3 replied:
There was meant to be another PR that would look inside of the inventory items.

OceanFish1 commented:
> > I'm a bit torn on items obtainable at ruins like toolboxes and ammo. They're not really worth the effort unless someone has a gimmick with illegal tech in mind, but I can still see the reasoning behind it so 🤷. Everything else is alright
> 
> You can get a syndicate hardsuit out of the space ruin with a safe, and that's definitely _worth it_ for illegal tech, since on one hand the hardsuit is nice by itself and very useful but on the other hand you can burn it for very useful illegal technology.

A syndie suit is too much for illegal tech nobody guaranteed that you will able to print illegal items after researching them. There're also miners tho

Glcbitj commented:
Stop saying "muh, not make sense red soap or red cards giff tech insight!1!!!", this game is not meant to be fucking realistic, if you haven't realized that yet, i suggest you to actually play it. Besides if you read said items description, you'll see they have a "nano-molecular edge" or some other lame ass excuse to explain their high technology, it's the reason why the soap cleans faster or something.

That being said, removing illegal tech from stuff you can commonly find in maints, roundstart gear and/or you can craft is probably fine, but removing it from ruins? that would make their reward significantly worse, ruins have really little cool stuff in them to begin with, you're just removing the utility of one of them. And flat out removing it from cheap Syndie Uplink is dumb, that and removing it from toolboxes would probably make illegal tech never be done ever again.

wraith-54321 commented:
I think that the ruin items should still give them as lets be honest, secs never gonna give up their power game traitor items, and I still want my donksoft

PowerfulBacon commented on code/_globalvars/lists/maintenance_loot.dm:
```diff
@@ -108,5 +108,6 @@ GLOBAL_LIST_INIT(maintenance_loot, list(
 	/obj/item/reagent_containers/food/snacks/canned/peaches/maint = 1,
 	/obj/item/reagent_containers/food/drinks/bottle/homemaderum = 1,
 	/obj/item/reagent_containers/food/snacks/grown/poppy = 10,
+	/obj/item/throwing_star = 1,
```
why add throwing stars to the loot pool?

bloons3 replied:
they no longer give illegal tech, and it is a single item

bloons3 committed:
remove inventory check
